elide-graphql/src/main/java/com/yahoo/elide/graphql/GraphQLConversionUtils.java

```diff
@@ -246,7 +335,13 @@ protected GraphQLOutputType attributeToQueryObject(Class<?> parentClass,
             }
 
             // If this is a collection of a boxed type scalar, we want to unwrap it properly
-            GraphQLOutputType outputType = fetchScalarOrObjectOutput(listType, ledger, fetcher);
+            GraphQLOutputType outputType;
+
+            if (listType.isEnum()) {
```

Comment:
Instead of all these checks for enums sprinkled throughout the file, we can maybe move the check for enum inside the fetchScalarOrObjectOutput (same comment applies for input objects as well).

Reply:
I agree. It would be worth refactoring to consolidate this logic.